### PR TITLE
Remove unused private methods

### PR DESCRIPTION
**Pull request by alamirault**
| Q             | A
| ------------- | ---
| Branch?       | 6.3
| Bug fix?      | no
| New feature?  | no 
| Deprecations? | no 
| Tickets       | Fix #... 
| License       | MIT
| Doc PR        | symfony/symfony-docs#... 

I searched all unused private methods in codebase and found these

### CODE REVIEW

File: src/Symfony/Component/Form/Tests/ChoiceList/Factory/DefaultChoiceListFactoryTest.php [hunk 1]
```diff
@@ -854,25 +854,6 @@ public function testCreateViewFlatlabelTranslationParametersClosureReceivesValue
         $this->assertFlatViewWithlabelTranslationParameters($view);
     }
 
-    private function assertScalarListWithChoiceValues(ChoiceListInterface $list)
```
Comment (alamirault) [inline src/Symfony/Component/Form/Tests/ChoiceList/Factory/DefaultChoiceListFactoryTest.php:857]: Was not removed when legacy tests were removed 

https://github.com/symfony/symfony/commit/31eb49863ee6d192a98414fe0e5eb3242db4641f#diff-18cef2a172697b433048f25490e0ccf5bec0b85c48b16040fb9fa3cd6bcfb4c9

File: src/Symfony/Component/VarExporter/LazyGhostTrait.php [hunk 2]
```diff
@@ -379,13 +379,4 @@ public function __destruct()
             parent::__destruct();
         }
     }
-
-    private function setLazyObjectAsInitialized(bool $initialized): void
```
Comment (nicolas-grekas) [inline src/Symfony/Component/VarExporter/LazyGhostTrait.php]: wrong removal, a private method on a trait can still be used by consumers

**Review by nicolas-grekas — COMMENTED** [review]
LGTM except the trait

File: src/Symfony/Component/VarExporter/LazyGhostTrait.php (hunk 2 above)
Comment (alamirault) [reply src/Symfony/Component/VarExporter/LazyGhostTrait.php]: I didn't know ! Thanks, it's reverted

**Comment by nicolas-grekas** [comment]
Thank you @alamirault.

**Pull request merged**